pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -56,7 +56,8 @@
     private static final Logger log = LoggerFactory.getLogger(PulsarClientImpl.class);
 
     private final ClientConfiguration conf;
-    private final HttpClient httpClient;
+    private HttpClient httpClient;
+    private boolean isHttpURI;
```

Comment:
We should abstract the `LookupService` to have 2 different implementation instead of relying on the flag.